src/test/ui/borrowck/issue-64453.rs

```diff
@@ -3,7 +3,6 @@ struct Value;
 
 static settings_dir: String = format!("");
 //~^ ERROR calls in statics are limited to constant functions
-//~| ERROR calls in statics are limited to constant functions
```

Comment:
So there's one error less here... which function is the remaining error from?
Can you add a `format_args!`, just to make sure that that is still rejected inside `const`?

Reply:
The remaining error is from the conversion of `fmt::Arguments` to `String`.

`format_args!("")` does in fact compile now, since `panic!("")` expands to that. Is there some way to make it only `const` when used through `panic!("")`?

Reply:
> format_args!("") does in fact compile now

I feared this might happen; we need to fix that (so please definitely add a test).

> Is there some way to make it only const when used through panic!("")?

const checking happens after macro expansion, so this will be a bit tricky, but I still think adding a special hack in [const checking](https://github.com/rust-lang/rust/tree/master/compiler/rustc_mir/src/transform/check_consts) is the best way here. @oli-obk would have a better idea for how to best to that; I hope they are back soon. :D